File: AcademicPlanBackend/src/main/java/com/bntu/fitr/poit/zholudev/diplom/entity/IndependentWorkForm.java
package com.bntu.fitr.poit.zholudev.diplom.entity;

import javax.persistence.*;
import javax.persistence.Entity;

@Entity
@Table(name = "independent_work_forms")
public class IndependentWorkForm {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String text;

    public IndependentWorkForm() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }
}

File: AcademicPlanFrontend/src/app/entity/independent-work-form.ts
export class IndependentWorkForm {
  id: number;
  text: string;

  static clone(independentWorkForm: IndependentWorkForm): IndependentWorkForm{
    let clonedIndependentWorkForm = new IndependentWorkForm();
    clonedIndependentWorkForm.id = independentWorkForm.id;
    clonedIndependentWorkForm.text = independentWorkForm.text;
    return clonedIndependentWorkForm;
  }
}
